### PR TITLE
Capistrano

### PR DESCRIPTION
**Pull request by tomecho**

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -10,11 +10,11 @@ group :production do
 end
 
 group :development do
-  gem 'capistrano', require: false
-  gem 'capistrano-pending', require: false
-  gem 'capistrano-bundler', require: false
-  gem 'capistrano-rails', require: false
-  gem 'capistrano-passenger', require: false
+  # deployment
```
Comment (werebus) [inline Gemfile]: You can leave the `require: false`s.  You don't need these gems in your load path to run the app.

File: Capfile [hunk 2]
```diff
@@ -1,2 +1,29 @@
...
+#   https://github.com/capistrano/bundler
+#   https://github.com/capistrano/rails
+#   https://github.com/capistrano/passenger
+#
+require 'capistrano/rvm'
+# require 'capistrano/rbenv'
+# require 'capistrano/chruby'
+require 'capistrano/bundler'
+require 'capistrano/rails'
+require 'capistrano/rails/assets'
```
Comment (werebus) [inline Capfile]: `capistrano/rails` already requires `capistrano/rails/assets` and `capistrano/rails/migrations`

File: config/deploy/deploy/staging.rb [hunk 3]
```diff
@@ -0,0 +1,45 @@
+# Simple Role Syntax
```
Comment (werebus) [inline config/deploy/deploy/staging.rb]: Might as well remove this stage if you aren't using it.

Comment (werebus) [inline config/deploy/deploy/staging.rb]: Oh, actually, Why is this in `config/deploy/deploy/`?

File: config/deploy.rb [hunk 4]
```diff
@@ -38,10 +43,7 @@
 
   after :restart, :clear_cache do
     on roles(:web), in: :groups, limit: 3, wait: 10 do
-      # Here we can do anything such as:
-      # within release_path do
-      #   execute :rake, 'cache:clear'
-      # end
+      execute :bundle, :exec, "rake db:migrate RAILS_ENV=production"
```
Comment (werebus) [inline config/deploy.rb]: This seems to be a bit of overkill (do you have more than 3 app servers?).  Also, I *think* capistrano-rails [does this by default](https://github.com/capistrano/rails/blob/master/lib/capistrano/tasks/migrations.rake#L31).

File: config/deploy.rb [hunk 5]
```diff
@@ -1,11 +1,16 @@
 # config valid only for current version of Capistrano
-lock '3.4.0'
+lock '3.3.5'
 
-set :application, 'my_app_name'
-set :repo_url, 'git@example.com:me/my_repo.git'
+set :application, 'aggressive-epsilon'
+set :repo_url, 'git@github.com:umts/aggressive-epsilon.git'
+set :user, 'rails'
```
Comment (werebus) [inline config/deploy.rb]: Ugh.  Generic logins 🤢

**Review by werebus — COMMENTED** [review]
Seems mostly fine.

File: config/deploy.rb (hunk 5 above)
Comment (tomecho) [inline config/deploy.rb]: the user that deploys is whoever is deploying, im not sure what that user does (i think runs it)

Comment (werebus) [inline config/deploy.rb]: I'm pretty sure you should remove this line then.

Comment (werebus) [inline config/deploy.rb]: Confirmed, `:user` is the SSH user:  http://capistranorb.com/documentation/advanced-features/properties/

Comment (tomecho) [inline config/deploy.rb]: strange, it asks for my password
image

File: config/deploy.rb (hunk 4 above)
Comment (tomecho) [inline config/deploy.rb]: it does actually, in fact this line is the root of most of my problems!

File: config/deploy/deploy/staging.rb (hunk 3 above)
Comment (tomecho) [inline config/deploy/deploy/staging.rb]: copying mistake

File: config/deploy.rb [hunk 6]
```diff
@@ -1,48 +1,15 @@
...
+set :application, 'aggressive-epsilon'
+set :repo_url, 'git@github.com:umts/aggressive-epsilon.git'
+set :deploy_to, '/var/www/aggressive-epsilon'
+set :rvm_type, :system
+set :linked_files, %w{config/database.yml config/secrets.yml}
+SSHKit.config.umask = '002'
+
+after :deploy, :finished do
+  on roles(:web) do
+    execute :rm, "-r /tmp/aggressive-epsilon/"
```
Comment (werebus) [inline config/deploy.rb]: What creates this directory?

Comment (tomecho) [inline config/deploy.rb]: It is created by the capistrano deploy process.  If i do not remove it (or change its permissions) multiple users are unable to deploy, capistrano will claim it cannot write to a file in that directory and fail.

Comment (werebus) [inline config/deploy.rb]: Ah.  I remember this now.  I fixed this [like so](https://github.com/umts/jobapps/blob/master/config/deploy/production.rb#L6).

**Pull request merged**